src/main/java/uk/co/ramp/covid/simulation/population/Places.java

```diff
@@ -0,0 +1,179 @@
+package uk.co.ramp.covid.simulation.population;
+
+import uk.co.ramp.covid.simulation.place.*;
+import uk.co.ramp.covid.simulation.util.RNG;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/** Helper class to manage communal places of particular types */
+public class Places {
+
+    private List<Office> offices;
+    private List<ConstructionSite> constructionSites;
+    private List<Hospital> hospitals;
+    private List<Nursery> nurseries;
+    private List<Restaurant> restaurants;
+    private List<School> schools;
+    private List<Shop> shops;
+
+    public List<Office> getOffices() {
+        return offices;
+    }
+
+    public List<ConstructionSite> getConstructionSites() {
+        return constructionSites;
+    }
+
+    public List<Hospital> getHospitals() {
+        return hospitals;
+    }
+
+    public List<Nursery> getNurseries() {
+        return nurseries;
+    }
+
+    public List<Restaurant> getRestaurants() {
+        return restaurants;
+    }
+
+    public List<School> getSchools() {
+        return schools;
+    }
+
+    public List<Shop> getShops() {
+        return shops;
+    }
+
+    public List<CommunalPlace> getAll() {
+        return all;
+    }
+
+    private List<CommunalPlace> all;
+
+    public Places() {
+        offices = new ArrayList<>();
+        constructionSites = new ArrayList<>();
+        hospitals = new ArrayList<>();
+        nurseries = new ArrayList<>();
+        restaurants = new ArrayList<>();
+        schools = new ArrayList<>();
+        shops = new ArrayList<>();
+        all = new ArrayList<>();
+    }
+
+    public void addOffice(Office o) {
+        offices.add(o);
+        all.add(o);
+    }
+
+    public void addConstructionSite(ConstructionSite s) {
+        constructionSites.add(s);
+        all.add(s);
+    }
+
+    public void addHospital(Hospital h) {
+        hospitals.add(h);
+        all.add(h);
+    }
+
+    public void addNursery(Nursery n) {
+        nurseries.add(n);
+        all.add(n);
+    }
+
+    public void addRestaurant(Restaurant r) {
+        restaurants.add(r);
+        all.add(r);
+    }
+
+    public void addSchool(School s) {
+        schools.add(s);
+        all.add(s);
+    }
+
+    public void addShop(Shop s) {
+        shops.add(s);
+        all.add(s);
+    }
+
+    // We often need a random place
+    private <T> T getRandom(List<T> s) {
+        int i = (int) RNG.get().nextInt(0, s.size() - 1);
+        return s.get(i);
```

Comment:
Should you check that s.size() > 0. It is (was?) possible to get no hospitals if the population was small.

Reply:
Good catch, I'll add a check and we can return null if there's nothing  (since this is what we need for the primaryPlaces if we don't get one). We might be able to handle this better in the future.